django/conf/__init__.py

```diff
@@ -179,6 +186,20 @@ def __init__(self, settings_module):
             os.environ['TZ'] = self.TIME_ZONE
             time.tzset()
 
+    def was_settings_module_found(self):
+        _, _, traceback = sys.exc_info()
+        # The first file in the traceback is this very file, so ignore it.
+        traceback = traceback.tb_next
+        while traceback:
+            # Continue stepping through frames as long as we're in the importlib module.
+            file = traceback.tb_frame.f_code.co_filename
+            if "importlib" not in file:
```

Comment:
Is there a better condition here that will match fewer file paths that accidentally say "importlib", e.g. `"importlib" in Path(file).parts` ?

Reply:
Hi @adamchainz, I've updated the code based on your recommendation. In addition to that, I also had to check for `file.startswith('<frozen importlib.'))`.